ocaml/xapi/xapi_pool.ml

```diff
@@ -2831,3 +2831,29 @@ let set_repository ~__context ~self ~value =
             Db.Pool.set_repository ~__context ~self ~value;
             Db.Repository.set_hash ~__context ~self:value ~value:"";
             Db.Repository.set_up_to_date ~__context ~self:value ~value:false))
+
+let sync_updates ~__context ~self ~force =
+  let open Repository in
+  match force with
+  | true ->
```

Comment:
`with_pool_operation` will be used by following APIs:
1. set_repository
2. get_updates: in next PR
3. sync_updates
4. apply_updates: in next of next PR

While in `sync_updates`, the `sync` may take long time, which may fail other APIs like `get_updates`.
The code here is to use another lock `reposync_mutex` to protect the sync between `set_repository` and `sync_updates`.  In other words, `get_updates` can be served even a `sync_updates` is in syncing.

Reply:
So operations like get_updates and apply_updates can run at the same time as `sync`, but not with `cleanup_repo` and `create_pool_repository`? And `with_reposync_lock` just exists so that you can take `sync` outside of `Xapi_pool_helpers.with_pool_operation`? But not in the `true` case, because you need to do a cleanup under with_pool_operation first anyway? Let's at least add a comment about this ;)

Reply:
Exactly. I will add more comments in next PR for this. Thanks.